src/main/java/lavalink/client/io/LavalinkRestClient.java

```diff
@@ -0,0 +1,168 @@
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+
+public final class LavalinkRestClient {
+
+    private static final ExecutorService EXECUTOR_SERVICE = Executors.newCachedThreadPool(new DaemonThreadFactory("Lavalink-RestExecutor"));
+    private static final Consumer<IOException> DEFAULT_ERROR_CONSUMER = Throwable::printStackTrace;
+
+    private final JdaLink link;
+    private Consumer<HttpClientBuilder> builderConsumer;
+
+    public LavalinkRestClient(final JdaLink link) {
+        this.link = link;
+    }
+
+    private LavalinkSocket getSocket() {
+        return link.getNode(true);
+    }
+
+    private String buildBaseAddress(final LavalinkSocket socket) {
+        return socket.getRemoteUri().toString().replace("ws://", "http://") + "/loadtracks?identifier=";
+    }
+
+    private HttpClient buildClient() {
+        final HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        if (builderConsumer == null)
+            return httpClientBuilder.build();
+        builderConsumer.accept(httpClientBuilder);
+        return httpClientBuilder.build();
+    }
+
+    private JSONObject apiGet(final String url, final String auth) throws IOException {
+        final HttpGet request = new HttpGet(url);
+        request.setHeader(HttpHeaders.AUTHORIZATION, auth);
+        final HttpClient httpClient = buildClient();
+        final HttpResponse httpResponse = httpClient.execute(request);
+        final int statusCode = httpResponse.getStatusLine().getStatusCode();
+        if (statusCode != 200)
+            throw new IOException("Invalid API Request Status Code: " + statusCode);
+        final HttpEntity entity = httpResponse.getEntity();
+        if (entity == null)
+            throw new IOException("Invalid API Response: No Content");
+        final String response = EntityUtils.toString(entity);
+        return new JSONObject(response);
+    }
+
+    public void setHttpClientBuilder(final Consumer<HttpClientBuilder> clientBuilder) {
+        this.builderConsumer = clientBuilder;
+    }
+
+    public void getYoutubeSearchResults(final String query, final Consumer<List<AudioTrack>> callback) {
+        getYoutubeSearchResults(query, callback, DEFAULT_ERROR_CONSUMER);
+    }
+
+    public void getYoutubeSearchResults(final String query, final Consumer<List<AudioTrack>> callback, final Consumer<IOException> errorCallback) {
+        EXECUTOR_SERVICE.submit(() -> {
+            try {
+                final List<AudioTrack> tracks = getYoutubeSearchResultsSync(query);
+                callback.accept(tracks);
+            } catch (final IOException ex) {
+                errorCallback.accept(ex);
+            }
+        });
+    }
+
+    public List<AudioTrack> getYoutubeSearchResultsSync(final String query) throws IOException {
```

Comment:
These search methods seem redundant and rather specific. Why not use the generic load method?